src/resolve-dependency.ts

```diff
@@ -50,7 +50,7 @@ function resolveDir (path: string, parent: string, job: Job) {
   if (pkgCfg && typeof pkgCfg.main === 'string') {
     const resolved = resolveFile(resolve(path, pkgCfg.main), parent, job) || resolveFile(resolve(path, pkgCfg.main, 'index'), parent, job);
     if (resolved) {
-      job.emitFile(path + sep + 'package.json', 'resolve', parent);
+      job.emitFile(job.realpath(path, parent) + sep + 'package.json', 'resolve', parent);
```

Comment:
```suggestion
      job.emitFile(job.realpath(path + sep + 'package.json', parent), 'resolve', parent);
```

This will handle package.json symlinks properly.